django/db/models/query.py

```diff
@@ -1331,7 +1331,8 @@ def _chain(self, **kwargs):
         if obj._sticky_filter:
             obj.query.filter_is_sticky = True
             obj._sticky_filter = False
-        obj.__dict__.update(kwargs)
+        if kwargs:
+            obj.__dict__.update(kwargs)
```

Comment:
As you say, this exists to support the following:

https://github.com/django/django/blob/8208381ba6a3d1613bb746617062ccf1a6a28591/django/db/models/query.py#L1626-L1629

But, as that is the only call that passes arguments, if we change that to:

```python
            queryset = self.queryset._chain()
            queryset._result_cache = []
            queryset._prefetch_done = True
            obj_dict['queryset'] = queryset
```

Then we can ditch this entirely:

```suggestion
```

...and drop the `**kwargs` argument from `QuerySet._chain()`.

Reply:
True enough, I realised a similar thing _after the fact_, but then didn't want to go back and change the implementation from what I was proposing on the ticket unduly. But now you're here, proposing same, and the ticket has been accepted, so I'll update it shortly :)